lms/create-concept-result-analyzer

anathomical opened the pull request:
## WHAT
Add a background worker to figure out which old_concept_ids didn't migrate
## WHY
So that we can get a list of items to try migrating again in order to hopefully fill in all of our gaps.  We want all of our `OldConceptResult` records to be successfully migrated.
## HOW
Do a series of LEFT OUTER JOINS to figure out which `OldConceptResult`s don't have a corresponding `ConceptResult` record, then email a list of all of those ids to Thomas for later use.

### Notion Card Links
https://www.notion.so/quill/ConceptResult-migration-and-model-swap-over-plan-1736cf7e688a4af79bab08116832f852

PR Checklist | Your Answer
------------ | -------------
Have you added and/or updated tests? |  Yes
Have you deployed to Staging? | Yes
Self-Review: Have you done an initial self-review of the code below on Github? | Yes
Spec Review: Have you reviewed the spec and ensured this meets requirements and/or matches design mockups? | N/A

anathomical committed:
Add a background worker to figure out which old_concept_ids didn't migrate

anathomical committed:
Address linting errors

anathomical commented on services/QuillLMS/app/mailers/application_mailer.rb:
```diff
@@ -0,0 +1,5 @@
+# frozen_string_literal: true
+
+class ApplicationMailer < ActionMailer::Base
```
We have a linting rule not to allow new classes to inherit from `Action____::Base`, so we need to create intermediate `Application____` models to inherit from.  This is the first `Mailer` that we've written since that rule got put into place.

brendanshean reviewed (approved):
Looks good!  Nice clear testing.

anathomical committed:
Make a small tweak to make ordering explicit

Pretty sure that the way we execute the code results in implicit ordering by ID, but it's worth making it clear that this is what's happening to future developers

anathomical committed:
Add a small spec for the SendAttachmentMailer
